From 4e31772eab4f39a5bfd6ae6c0ad192cf1299456f Mon Sep 17 00:00:00 2001
From: Chih-Hsuan Yen <yan12125@gmail.com>
Date: Sun, 23 Jun 2019 21:22:45 +0800
Subject: [PATCH] Fix a flaky test

test_config_expand_paths sometimes fails with:

======================================================================
FAIL: test_config_expand_paths (module.config.test_config_mixin.TestConfigMixIn)
Test precedence in
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yen/var/syncthing/Projects/cfn-python-lint/.tox/py37/lib/python3.7/site-packages/mock/mock.py", line 1330, in patched
    return func(*args, **keywargs)
  File "/home/yen/var/syncthing/Projects/cfn-python-lint/test/module/config/test_config_mixin.py", line 97, in test_config_expand_paths
    'test/fixtures/templates/public/rds-cluster.yaml'])
AssertionError: Lists differ: ['tes[23 chars]blic/rds-cluster.yaml', 'test/fixtures/templat[25 chars]aml'] != ['tes[23 chars]blic/lambda-poller.yaml', 'test/fixtures/templ[25 chars]aml']

First differing element 0:
'test/fixtures/templates/public/rds-cluster.yaml'
'test/fixtures/templates/public/lambda-poller.yaml'

- ['test/fixtures/templates/public/rds-cluster.yaml',
-  'test/fixtures/templates/public/lambda-poller.yaml']
? ^                                                   ^

+ ['test/fixtures/templates/public/lambda-poller.yaml',
? ^                                                   ^

+  'test/fixtures/templates/public/rds-cluster.yaml']

----------------------------------------------------------------------
---
 src/cfnlint/config.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cfnlint/config.py b/src/cfnlint/config.py
index 5b16a231..9d7debc2 100644
--- a/src/cfnlint/config.py
+++ b/src/cfnlint/config.py
@@ -545,7 +545,7 @@ def templates(self):
                     if add_filename not in ignore_templates:
                         all_filenames.append(add_filename)
 
-        return all_filenames
+        return sorted(all_filenames)
 
     def _ignore_templates(self):
         """ templates """
